src/CarbonAware.DataSources/CarbonAware.DataSources.Json/src/Configuration/JsonDataConfiguration.cs

```diff
@@ -0,0 +1,49 @@
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace CarbonAware.DataSources.Json.Configuration;
+
+/// <summary>
+/// A configuration class for holding Json Data config values.
+/// </summary>
+public class JsonDataConfiguration
+{
+    private const string BaseDir = "data-files";
+    private const string DefaultDataFile = "test-data-azure-emissions.json";
+    private const string RegExDir = @"^[-/a-zA-Z_\d ]*$";
+    private string? dataFileLocation;
+    private string assemblyDirectory;
+
+    public const string Key = "JsonData";
+
+
+    public JsonDataConfiguration()
+    {
+        var assemblyPath = Assembly.GetExecutingAssembly().Location;
+        assemblyDirectory = Path.GetDirectoryName(assemblyPath)!;
+        DataFileLocation = DefaultDataFile;
+    }
+
+    /// <summary>
+    /// Json data file location
+    /// </summary>
+    public string DataFileLocation
+    {
+        get => dataFileLocation!;
+        set
+        {
+            if (!IsValidDirPath(value))
+            {
+                throw new ArgumentException($"File path '{value}' contains not supported characters.");
+            }
+            dataFileLocation = Path.Combine(assemblyDirectory, BaseDir, value);
+        }
+    }
+
+    private static bool IsValidDirPath(string fileName)
+    {
```

Comment:
I think with this Regex, it will allow blank fileName. So, if the fileName is not configured, will this break?

Reply:
By default is configured, unless you set it to something else at start time. This method only gets the dir part of the string, meaning whatever is last before the dir separator. If the file doesn't exists, it is fine, there would be an error. The goal of this regex is to avoid to travers the filesystem to get to places you don't want to be. It is a sandbox basically under 'data-files' dir. Not a check for file existence.